include/hip/hcc_detail/functional_grid_launch.hpp

```diff
@@ -37,14 +37,15 @@ THE SOFTWARE.
 hipError_t ihipExtLaunchMultiKernelMultiDevice(hipLaunchParams* launchParamsList, int numDevices,
                                                unsigned int flags, hip_impl::program_state& ps);
 
-hipError_t ihipLaunchCooperativeKernel(const void* f, dim3 gridDim, dim3 blockDimX, void** kernelParams,
-                unsigned int sharedMemBytes, hipStream_t stream, hip_impl::program_state& ps);
-
-hipError_t ihipLaunchCooperativeKernelMultiDevice(hipLaunchParams* launchParamsList, int  numDevices,
-                unsigned int  flags, hip_impl::program_state& ps);
-
-
+hipError_t hipLaunchCooperativeKernel(const void* f, dim3 gridDim,
+                                    dim3 blockDim, void** args,
+                                    size_t sharedMem, hipStream_t stream,
+                                    hip_impl::program_state& ps);
```

Comment:
I think we should remove **hip_impl::program_state& ps** in order to match it with CUDA API signature

Reply:
@SarbojitAMD the hipLaunchCooperativeKernel here is an overloaded internal function which is used to call the internal  ihipLaunchCooperativeKernel function. The hip_impl::program_state& ps is needed here and should not be removed.

Reply:
Agreed with @asalmanp for these. These overloaded functions exist as trampolines to reach the internal cooperative launch functions. ihipLogStatus() explicitly says we should not log the status of internal functions (so don't use it in ihip* locations), so these functions exist as a safe place to do HIP_INIT() and ihipLogStatus() before diving into the ihip cooperative launch functions.

As Aryan notes, `ps` is needed inside the ihip cooperative launch function, so we're passing it through here.

The actual user-visible hipLaunchCooperativeKernel() without `hip_impl::program_state& ps` still exists.